src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts

```diff
@@ -179,7 +179,10 @@ export class XtermTerminal extends Disposable implements IXtermTerminal, IDetach
 	}
 
 	public get isFocused() {
-		return !!this.raw.element?.contains(document.activeElement);
+		if (!this.raw.element) {
+			return false;
+		}
+		return dom.isAncestorOfActiveElement(this.raw.element);
```

Comment:
TIL about `node.contains`, I think I can use that to simplify the implementation of the `dom` utility.